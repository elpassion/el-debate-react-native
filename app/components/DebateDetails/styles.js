import { StyleSheet, Platform } from 'react-native'

export default StyleSheet.create({
  navContainer: {
    flex: 1
  },
  container: {
    flex: 1,
    top: (Platform.OS === 'ios') ? 85 : 55,
  },
  description: {
    marginLeft: 20,
    marginBottom: 20,
    fontSize: 12,
    fontFamily: 'helvetica-neue',
    color: '#B8B8B8',
    backgroundColor: 'transparent'
  },
  topicBox: {
    alignSelf: 'center',
    justifyContent: 'center',
    width: '90%',
    height: 100,
    borderColor: 'gray',
    borderRadius: 5,
    borderStyle: 'solid',
    borderWidth: 0.3,
    marginBottom: 20
  },
  topic: {
    alignSelf: 'center',
    fontSize: 24,
    fontFamily: 'helvetica-neue',
    color: '#4A4A4A'
  },
  answerBox: {
    alignSelf: 'center',
    justifyContent: 'center',
    width: '90%',
    height: 60,
    backgroundColor: 'white',
    borderColor: 'gray',
    borderRadius: 5,
    borderStyle: 'solid',
    borderWidth: 0.3,
    marginBottom: 20
  },
  positiveAnswer: {
    marginLeft: 20,
    fontSize: 24,
    fontFamily: 'helvetica-neue',
    color: '#0098E3'
  },
  negativeAnswer: {
    marginLeft: 20,
    fontSize: 24,
    fontFamily: 'helvetica-neue',
    color: '#E44043'
  },
  neutralAnswer: {
    marginLeft: 20,
    fontSize: 24,
    fontFamily: 'helvetica-neue',
    color: '#8F8F8F'
  },
  footer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0
  },
  backgroundImage: {
    alignSelf: 'center',
    bottom: -30
  }
})
