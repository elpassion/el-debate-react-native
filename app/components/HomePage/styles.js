import { StyleSheet, Platform } from 'react-native'

export default styles = StyleSheet.create({
  description: {
    marginBottom: 20,
    fontSize: 20,
    textAlign: 'center',
    fontFamily: 'helvetica-neue',
    color: '#656565'
  },
  navContainer: {
    flex: 1
  },
  container: {
    alignItems: 'center',
    flex: 1,
    top: (Platform.OS === 'ios') ? 100 : 70,
  },
  backgroundImage: {
    position: 'absolute',
    bottom: (Platform.OS === 'ios') ? 50 : 0,
    width: '85%',
  },
  iosPinInput: {
    bottom: 0,
    height: 40,
    width: '80%',
    textAlign: 'center',
    marginRight: 5,
    marginTop: 40,
    borderBottomColor: '#C0C0C0',
    borderBottomWidth: 1,
    fontSize: 18,
    color: '#8F8F8F',
  },
  androidPinInput: {
    bottom: 0,
    height: 60,
    width: '80%',
    textAlign: 'center',
    marginRight: 5,
    marginTop: 40,
    fontSize: 20,
    color: '#8F8F8F',
  },
  pinButton: {
    alignSelf: 'center',
    height: (Platform.OS === 'ios') ? 40 : 50,
    width: '80%',
    bottom: (Platform.OS === 'ios') ? 120 : 80,
    backgroundColor: '#4CC359',
    borderRadius: 0,
    borderColor: '#fff',
  },
  footer: {
    position: 'absolute',
    bottom: (Platform.OS === 'ios') ? 0 : 50,
    right: 0,
    left: 0,
    alignItems: 'center'
  }
})
