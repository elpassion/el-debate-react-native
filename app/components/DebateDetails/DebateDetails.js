import React, { Component } from 'react'
import {
  StyleSheet,
  Image,
  View,
  TouchableHighlight,
  FlatList,
  Text,
  AsyncStorage,
  ScrollView,
  TouchableOpacity,
  BackHandler,
} from 'react-native';
import Button from 'apsl-react-native-button'
import styles from './styles'
import Api from '/app/api/Api'
import AlertModal from '/app/components/Shared/AlertModal'
import * as Progress from 'react-native-progress';

export default class DebateDetails extends Component {
  constructor(props) {
    super(props)

    this.state = {
      authToken: '',
      topic: '',
      positiveAnswer: '',
      neutralAnswer: '',
      negativeAnswer: '',
      positiveAnswerId: 0,
      negativeAnswerId: 0,
      neutralAnswerId: 0,
      lastAnswerId: 0,
      isLoading: false,
      isFetched: false,
      answerPressed: false,
      clearId: 0
    }

    AsyncStorage.getItem('authToken')
      .then((value) => this.getDebate(value))
  }

  getDebate = (authToken) => {
    this.setState({ authToken: authToken })
    Api.getDebate(authToken)
      .then(response => response.json())
      .then(response => this._handleDebateResponse(response))
      .catch(error =>
        this.refs.alertModal.openModalAlert(error)
      );
  }

  _handleDebateResponse = (response) => {
    this.setState({
      topic: response.topic,
      positiveAnswer: response.answers.positive.value,
      negativeAnswer: response.answers.negative.value,
      neutralAnswer: response.answers.neutral.value,
      positiveAnswerId: response.answers.positive.id,
      negativeAnswerId: response.answers.negative.id,
      neutralAnswerId: response.answers.neutral.id,
      lastAnswerId: response.last_answer_id,
      isFetched: true
    })
  }

  _onPressAnswer = (answerId) => {
    this.setState({ isLoading: true, answerPressed: true })
    Api.vote(this.state.authToken, answerId)
      .then(response => this._handleVote(response, answerId))
      .catch(error =>
        this.refs.alertModal.openModalAlert(error)
      );
  }

  _handleVote = (response, answerId) => {
    if (response.status === 201) {
      this.setState({ lastAnswerId: answerId })
      this.refs.alertModal.openModalAlert('Vote has been saved')
    } else if (response.status === 403) {
      this.refs.alertModal.openModalAlert('Debate is closed.')
    }
    var clearId = setTimeout(() => { this.setState({ answerPressed: false }) }, 500);
    this.setState({ isLoading: false, clearId: clearId })
  }

  componentWillMount() {
    BackHandler.addEventListener('hardwareBackPress', function() {
      this.props.navigation.goBack()
      return true
    }.bind(this))
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress')
    clearTimeout(this.state.clearId)
  }

  render() {
    const PositiveIcon =
      this.state.lastAnswerId === this.state.positiveAnswerId ?
        require('/resources/images/PositiveIcon.png') :
        require('/resources/images/PositiveIconGrey.png')
    const NegativeIcon =
      this.state.lastAnswerId === this.state.negativeAnswerId ?
        require('/resources/images/NegativeIcon.png') :
        require('/resources/images/NegativeIconGrey.png')
    const NeutralIcon =
      this.state.lastAnswerId === this.state.neutralAnswerId ?
        require('/resources/images/NeutralIcon.png') :
        require('/resources/images/NeutralIconGrey.png')

    return (
      <View style={styles.navContainer}>
        <View style={styles.container}>
          <ScrollView style={styles.main}>
            <Text style={styles.description}>
              Our debate is about:
            </Text>
            <View style={styles.topicBox}>
              <Text style={styles.topic}>
                {this.state.topic}
              </Text>
            </View>
            <Text style={[styles.description, { marginBottom: 0 }]}>
              Choose your side with one of the following:
            </Text>
            <TouchableOpacity
              disabled={this.state.answerPressed || !this.state.isFetched }
              style={styles.answerBox}
              onPress={() => this._onPressAnswer(this.state.positiveAnswerId)}>
              <Text style={styles.positiveAnswer}>
                {this.state.positiveAnswer}
              </Text>
              <Image
                source={PositiveIcon}
                style={styles.positiveAnswerIcon}/>
            </TouchableOpacity>
            <View style={styles.progressBar}>
              <Progress.Bar progress={0} color={'#0098E3'} height={2} width={null} indeterminate={false} borderWidth={0}/>
            </View>
            <TouchableOpacity
              disabled={this.state.answerPressed || !this.state.isFetched }
              style={styles.answerBox}
              onPress={() => this._onPressAnswer(this.state.negativeAnswerId)}>
              <Text style={styles.negativeAnswer}>
                {this.state.negativeAnswer}
              </Text>
              <Image
                source={NegativeIcon}
                style={styles.negativeAnswerIcon}/>
            </TouchableOpacity>
            <View style={styles.progressBar}>
              <Progress.Bar progress={0} color={'#E44043'} height={2} width={null} indeterminate={false} borderWidth={0}/>
            </View>
            <TouchableOpacity
              disabled={this.state.answerPressed || !this.state.isFetched }
              style={styles.answerBox}
              onPress={() => this._onPressAnswer(this.state.neutralAnswerId)}>
              <Text style={styles.neutralAnswer}>
                {this.state.neutralAnswer}
              </Text>
              <Image
                source={NeutralIcon}
                style={styles.neutralAnswerIcon}/>
            </TouchableOpacity>
            <View style={styles.progressBar}>
              <Progress.Bar progress={0} color={'#8F8F8F'} height={2} width={null} indeterminate={false} borderWidth={0}/>
            </View>
            <Text style={[styles.description, { marginTop: 20 }]}>
              Remember that you can change your mind before debate ends, thats why we are here!
            </Text>
          </ScrollView>
          <View style={styles.footer}>
            <Image source={require('/resources/images/backgroundimg.png')} style={styles.backgroundImage} />
            <Button isDisabled={!this.state.isFetched} onPress={() => this.props.navigation.navigate('CommentsList')} style={styles.button} title="Chat">
              <Text style={styles.buttonText}>
                Chat
              </Text>
            </Button>
          </View>
        </View>
        <AlertModal
          ref="alertModal"
        />
      </View>
    );
  }
}
