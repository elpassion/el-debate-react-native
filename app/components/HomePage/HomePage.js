import React, { Component } from 'react';
import {
  Text,
  TextInput,
  View,
  ActivityIndicator,
  Image,
  Alert,
  Platform,
  AsyncStorage
} from 'react-native';
import Button from 'apsl-react-native-button'
import styles from './styles'
import { Redirect, Route } from "react-router-native";
import NavigationBar from 'react-native-navigation-bar';
import Api from '/app/api/Api'

export default class HomePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      message: '',
      searchString: '',
      isFetched: false
    };
  }

  _handleResponse = (response) => {
    if (response.error) {
      this.setState({ message: response.error });
    } else {
      this.setState({ isFetched: true })
      AsyncStorage.setItem('authToken', response.auth_token)
    }
  };

  _onLoginPressed = () => {
    this.setState({ isLoading: true, message: '' });
    AsyncStorage.setItem('code', this.state.searchString)
    Api.login(this.state.searchString)
      .then(response => response.json())
      .then(response => this._handleResponse(response))
      .catch(error =>
        Alert.alert('Something went wrong' + error)
    );
    this.setState({ isLoading: false });
  };

  _onSearchTextChanged = (event) => {
    this.setState({ searchString: event.nativeEvent.text });
  };

  componentDidMount = () => {
     AsyncStorage.getItem('code').then((value) => this.setState({ 'searchString': value }))
  }

  render() {
    const DetailsRoute = () => (
      this.state.isFetched ? <Redirect push to="/debate-details"/> : null
    )
    const CodeInput =
      this.state.isLoading ?
      <View style={{marginTop: 40}}>
        <ActivityIndicator size='large'/>
      </View> :
      <TextInput
        keyboardType = {(Platform.OS === 'ios') ? 'default' : 'numeric'}
        maxLength = {5}
        style={(Platform.OS === 'ios') ? styles.iosPinInput : styles.androidPinInput}
        value={this.state.searchString}
        onChange={this._onSearchTextChanged}
        placeholder='EL Debate PIN'
        returnKeyType = 'done'
        onSubmitEditing={this._onLoginPressed}/>;
    return (
      <View style={styles.navContainer}>
        <NavigationBar
          title={'EL Debate'}
          height={50}
          titleColor={'#fff'}
          backgroundColor={'#4CC359'}
        />
      <View style={styles.container}>
        <Text style={styles.description}>{this.state.message}</Text>
        <Text style={styles.description}>
          Welcome to
        </Text>
        <Image source={require('/resources/images/logo.png')}/>
        {CodeInput}
        <View style={styles.footer}>
          <Image source={require('/resources/images/backgroundimg.png')} style={styles.backgroundImage} />
          <Button
            onPress={this._onLoginPressed}
            style={styles.pinButton}
            textStyle={{ color: '#fff', fontFamily: 'helvetica-neue', fontSize: 14 }}
          >
          Log in
          </Button>
          <DetailsRoute/>
        </View>
        </View>
      </View>
    );
  }
}
