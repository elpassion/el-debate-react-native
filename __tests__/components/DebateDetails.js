import 'react-native';
import React from 'react';
import HomePage from 'app/components/HomePage/HomePage';
import DebateDetails from 'app/components/DebateDetails/DebateDetails'

it('renders correctly', () => {
  const wrapper = shallow(<DebateDetails />)
  expect(wrapper).toMatchSnapshot();
});

it('renders debate details component if isFetched is true on HomePage', () => {
  const wrapper = shallow(<HomePage isFetched={false}/>)
  expect(wrapper).toMatchSnapshot();
  wrapper.setProps({ isFetched: true })
  expect(wrapper).toMatchSnapshot('DebateDetails');
});
