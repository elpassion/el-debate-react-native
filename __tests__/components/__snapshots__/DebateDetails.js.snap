// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebateDetails 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <NavigationBar
    backgroundColor="#4CC359"
    height={50}
    leftButtonTitle={null}
    leftButtonTitleColor="#000"
    rightButtonTitle={null}
    rightButtonTitleColor="#000"
    title="EL Debate"
    titleColor="#fff"
  />
  <View
    style={
      Object {
        "flex": 1,
        "top": 100,
      }
    }
  >
    <ScrollViewMock
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#656565",
            "fontFamily": "helvetica-neue-lt",
            "fontSize": 20,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#656565",
            "fontFamily": "helvetica-neue-lt",
            "fontSize": 20,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        Welcome to
      </Text>
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 50,
            "resizeMode": "contain",
            "width": 106,
          }
        }
      />
      <View>
        <TextInput
          keyboardType="numeric"
          maxLength={5}
          onChange={[Function]}
          onSubmitEditing={[Function]}
          placeholder="EL Debate PIN"
          returnKeyType="done"
          style={
            Object {
              "alignSelf": "center",
              "borderBottomColor": "#C0C0C0",
              "borderBottomWidth": 1,
              "bottom": 0,
              "color": "#8F8F8F",
              "fontSize": 18,
              "height": 40,
              "marginRight": 5,
              "marginTop": 40,
              "textAlign": "center",
              "width": "80%",
            }
          }
          value=""
        />
        <Image
          source={1}
          style={
            Object {
              "bottom": 30,
              "left": 45,
              "position": "relative",
            }
          }
        />
      </View>
    </ScrollViewMock>
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "bottom": 0,
          }
        }
      />
      <Button
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#4CC359",
            "borderColor": "#fff",
            "borderRadius": 0,
            "bottom": 120,
            "height": 40,
            "width": "80%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "helvetica-neue-lt",
              "fontSize": 14,
            }
          }
        >
          Log in
        </Text>
      </Button>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <NavigationBar
    backgroundColor="#4CC359"
    height={50}
    leftButtonTitle="Back"
    leftButtonTitleColor="#fff"
    onLeftButtonPress={[Function]}
    rightButtonTitle={null}
    rightButtonTitleColor="#000"
    title="EL Debate"
    titleColor="#fff"
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "bottom": -30,
        }
      }
    />
  </View>
  <ScrollViewMock
    style={
      Object {
        "flex": 1,
        "top": 85,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#B8B8B8",
          "fontFamily": "helvetica-neue-md",
          "fontSize": 12,
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      Our debate is about:
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "rgba(0, 0, 0, 0.3)",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 0.08,
          "height": 120,
          "justifyContent": "center",
          "marginBottom": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 2,
          "width": "90%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#4A4A4A",
            "fontFamily": "helvetica-neue-md",
            "fontSize": 24,
            "marginLeft": 20,
            "marginRight": 20,
          }
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#B8B8B8",
          "fontFamily": "helvetica-neue-md",
          "fontSize": 12,
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      Choose your side with one of the following:
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "rgba(0, 0, 0, 0.3)",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 0.08,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "marginBottom": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 2,
          "width": "90%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#0098E3",
            "fontFamily": "helvetica-neue-lt",
            "fontSize": 24,
            "marginLeft": 20,
            "marginTop": 15,
          }
        }
      />
      <Image
        source={1}
        style={
          Object {
            "marginRight": 20,
            "marginTop": 15,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "rgba(0, 0, 0, 0.3)",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 0.08,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "marginBottom": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 2,
          "width": "90%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#E44043",
            "fontFamily": "helvetica-neue-lt",
            "fontSize": 24,
            "marginLeft": 20,
            "marginTop": 15,
          }
        }
      />
      <Image
        source={1}
        style={
          Object {
            "marginRight": 20,
            "marginTop": 17,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderColor": "rgba(0, 0, 0, 0.3)",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 0.08,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "marginBottom": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 2,
          "width": "90%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#8F8F8F",
            "fontFamily": "helvetica-neue-lt",
            "fontSize": 24,
            "marginLeft": 20,
            "marginTop": 15,
          }
        }
      />
      <Image
        source={1}
        style={
          Object {
            "marginRight": 20,
            "marginTop": 15,
          }
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#B8B8B8",
          "fontFamily": "helvetica-neue-md",
          "fontSize": 12,
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      Remember that you can change your mind before debate ends, thats why we are here!
    </Text>
  </ScrollViewMock>
</View>
`;

exports[`renders debate details component if isFetched is true on HomePage 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <NavigationBar
    backgroundColor="#4CC359"
    height={50}
    leftButtonTitle={null}
    leftButtonTitleColor="#000"
    rightButtonTitle={null}
    rightButtonTitleColor="#000"
    title="EL Debate"
    titleColor="#fff"
  />
  <View
    style={
      Object {
        "flex": 1,
        "top": 100,
      }
    }
  >
    <ScrollViewMock
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#656565",
            "fontFamily": "helvetica-neue-lt",
            "fontSize": 20,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#656565",
            "fontFamily": "helvetica-neue-lt",
            "fontSize": 20,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        Welcome to
      </Text>
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 50,
            "resizeMode": "contain",
            "width": 106,
          }
        }
      />
      <View>
        <TextInput
          keyboardType="numeric"
          maxLength={5}
          onChange={[Function]}
          onSubmitEditing={[Function]}
          placeholder="EL Debate PIN"
          returnKeyType="done"
          style={
            Object {
              "alignSelf": "center",
              "borderBottomColor": "#C0C0C0",
              "borderBottomWidth": 1,
              "bottom": 0,
              "color": "#8F8F8F",
              "fontSize": 18,
              "height": 40,
              "marginRight": 5,
              "marginTop": 40,
              "textAlign": "center",
              "width": "80%",
            }
          }
          value=""
        />
        <Image
          source={1}
          style={
            Object {
              "bottom": 30,
              "left": 45,
              "position": "relative",
            }
          }
        />
      </View>
    </ScrollViewMock>
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "bottom": 0,
          }
        }
      />
      <Button
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#4CC359",
            "borderColor": "#fff",
            "borderRadius": 0,
            "bottom": 120,
            "height": 40,
            "width": "80%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "helvetica-neue-lt",
              "fontSize": 14,
            }
          }
        >
          Log in
        </Text>
      </Button>
    </View>
  </View>
</View>
`;
